taskcat/_cli_modules/test.py

```diff
@@ -31,6 +32,132 @@ class Test:
+        :param project_root: root path of the project relative to input_file
+        :param enable_sig_v2: enable legacy sigv2 requests for auto-created buckets
+        :param minimal_output: Reduces output during test runs
+        """
+
+        self.uid = uuid.uuid4()
+        self.test_names = test_names
+        self.regions = regions
+
+        # Temporary attributes that need removed
+        self.minimal_output = minimal_output
+
+        project_root_path: Path = Path(project_root).expanduser().resolve()
+        input_file_path: Path = project_root_path / input_file
+
+        args = _build_args(enable_sig_v2, regions, GLOBAL_ARGS.profile)
+
+        self.config = Config.create(
+            project_root=project_root_path,
+            project_config_path=input_file_path,
+            args=args
+            # TODO: detect if input file is taskcat config or CloudFormation template
+        )
+
+    def create_stacks(self, skip_upload: bool = False, lint_disable: bool = False):
+        """Creates the resources and stacks needed for testing
+
+        :param skip_upload: dont upload resources to s3
+        :param lint_disable: disable cfn-lint checks
+        """
+
+        _trim_regions(self.regions, self.config)
+        _trim_tests(self.test_names, self.config)
+
+        boto3_cache = Boto3Cache()
+
+        templates = self.config.get_templates()
+
+        if skip_upload and not self.config.config.project.s3_bucket:
+            raise TaskCatException(
+                "cannot skip_buckets without specifying s3_bucket in config"
+            )
+
+        buckets = self.config.get_buckets(boto3_cache)
+
+        if not skip_upload:
+            # 1. lint
+            if not lint_disable:
+                lint = TaskCatLint(self.config, templates)
+                errors = lint.lints[1]
+                lint.output_results()
+                if errors or not lint.passed:
+                    raise TaskCatException("Lint failed with errors")
+            # 2. build lambdas
+            if self.config.config.project.package_lambda:
+                LambdaBuild(self.config, self.config.project_root)
+            # 3. s3 sync
+            stage_in_s3(
+                buckets, self.config.config.project.name, self.config.project_root
+            )
+        # 4. launch stacks
+        regions = self.config.get_regions(boto3_cache)
+        parameters = self.config.get_rendered_parameters(buckets, regions, templates)
+        tests = self.config.get_tests(templates, regions, buckets, parameters)
+
+        self.test_definition = Stacker(
+            self.config.config.project.name,
+            tests,
+            shorten_stack_name=self.config.config.project.shorten_stack_name,
+        )
+        self.test_definition.create_stacks()
+        terminal_printer = TerminalPrinter(minimalist=self.minimal_output)
+        # 5. wait for completion
+        terminal_printer.report_test_progress(stacker=self.test_definition)
```

Comment:
afaik the only way to wait for stack completion is to use the terminal_printer. Which will need to change for testing without cli. 

I seen the `Stacker` class has a status method that has an argument `recurse` that is not implemented. I could probably implement it and wait for completion that way. 

My plan is to have the cli methods like `run` and `retry` to create their own termina_printer and pass them into the create and delete methods. So this would change to something like...

```suggestion
        if terminal_printer:  
        terminal_printer.report_test_progress(stacker=self.test_definition)
```
github wouldn't line up the white space ^